Vivekananda_session3_4VP20CS062_Prajwal/Batman/web_scrapper.py

```diff
@@ -0,0 +1,103 @@
+import requests
+import re
+from bs4 import BeautifulSoup, element
+import psycopg2
+
+# For the credentials mentioned below, you may refer the docker-compose.yml present in myworld .
+db_name = 'member_db2'
+db_user = 'postgres'
+db_pass = '123456'
+db_host = 'psql-db2.0'
+db_port = '5432'
+        
+# This will create the connection the to postgres database.
+conn = psycopg2.connect(dbname=db_name, user=db_user, password=db_pass, host=db_host, port=db_port)
+        
+        
+def add_row_to_blog(title, date, time, author, content, recommended, html):
+    # This function will add the entry to database
+    sql = """INSERT INTO members_blog (title, release_date, blog_time, author,created_date,  content, recommended, html) VALUES (%s, %s::DATE,    		 %s::TIME, %s, NOW(), %s, %s, %s)"""
+            
+    with conn:
+        with conn.cursor() as curs:
+            time=time.replace('\u202f',"")
+            curs.execute(sql, (title, date, time, author, content, recommended, html))
+        
+        
+def truncate_table():
+    # This function will delete the existing entries from the database.
+    with conn:
+        with conn.cursor() as curs:
+            curs.execute("TRUNCATE members_blog CASCADE;")
+        
+
+def start_extraction():
+    print("Extraction started")
+    url = "https://blog.python.org/"
+    data = requests.get(url)
+    page_soup = BeautifulSoup(data.text, 'html.parser')
+
+    blogs = page_soup.select('div.date-outer')
+
+    for blog in blogs:
+        date = blog.select('span', class_='date-header')[0].get_text()
+
+        post = blog.select('.post')[0]
+       # print(post)
+       # post_time = blog.find('abbr', class_='published')
+       
+       # title = blog.h3.a.text
+        #print(title)
+
+        title = ""
+        title_bar = post.select('.post-title')
+        if len(title_bar) > 0:
+            title = title_bar[0].text
+        else:
+            title = post.select('.post-body')[0].contents[0].text
+
+        # getting the author and blog time
+        post_footer = post.select('.post-footer')[0]
+
+        author = post_footer.select('.post-author span')[0].text
+
+        time = post_footer.select('abbr')[0].text
+
+        post_header = post.select('.post-header')[0]
+        
+        '''contents = page_soup.find_all('div',class_='date-posts')
+        #print(contents)
+        for con in contents:'''
+        content = post.select('.post-body')[0].text
+        html = 'S:\web_scapping\python_blogs.html'
+        with open(html, 'w', encoding='utf-8') as f:
```

Comment:
@Prajwal7Amin You're writing to a file inside the loop. Do you think your file will contain the whole html?

Reply:
@Prajwal7Amin in the `open` method, the first argument is the file name, what does that second argument indicate/ why is that used here?

Reply:
@anushds Second argument indicates the mode in which the file is opened. In this case w is the write mode.
It is used here to write the new data into the  file (html = 'S:\web_scraping\python_blogs.html').

Reply:
@Prajwal7Amin Alright, so you've used write mode here.
Do you know what are modes are available? Could you brief about them.

Reply:
@anushds Should I use append mode 'a' here instead of write mode?. 
Is it right? Can you help me with this.

Reply:
@anushds Should I use append mode 'a' here instead of write mode. Is it right. Can you help me with this.

Reply:
@Prajwal7Amin Good that you realised.
There are multiple approaches to this,

1.  Store the html contents of each iteration into a variable (say `html_contents'). After the loop ends write this into the file.
2. Yea, you can open the file in append mode and then write to it. But I wouldn't suggest doing that because opening and closing file every iteration is a expensive operation (provided the number of calls made to OS). Sure the difference will be in seconds but even these "extra " seconds consumed by your code in doing this operation matters from a production perspective.